lib/src/services/courseApi.dart

```diff
@@ -1,44 +1,57 @@
-import 'dart:convert';
-import 'dart:developer';
 import 'dart:io';
 
+import 'package:dio/dio.dart';
+import 'package:dio_http_cache/dio_http_cache.dart';
+import 'package:flutter/cupertino.dart';
 import 'package:oneplace_illinois/src/misc/config.dart';
 import 'package:oneplace_illinois/src/models/courseItem.dart';
 import 'package:oneplace_illinois/src/models/sectionItem.dart';
-import 'package:oneplace_illinois/src/services/api.dart';
 
 class CourseAPI {
-  Future<List<CourseItem>?> getCourses(ApiService api, String query,
+  final Dio dio = Dio();
+
+  Future<List<CourseItem>?> getCourses(String query,
       {bool onlyCourses = false}) async {
     Uri uri = Uri.http(Config.baseEndpoint!, '/api/v1/course/search',
         {"keyword": query, "only_courses": onlyCourses.toString()});
-    final response = await api.get(uri);
+    dio.interceptors.add(
+      DioCacheManager(
+        CacheConfig(
+          baseUrl: Config.baseEndpoint!,
+          defaultMaxAge: Duration(
+            days: 1,
```

Comment:
Do you know if there's a way to invalidate the cache here? If not, I say we make this only ~1hr and use it solely for unauthenticated requests. That way, if someone subscribes to a new section and receives a feed item (or otherwise receives one), they'll get it instantly.

Reply:
Yeah, we can invalidate caches, iirc. I haven't looked too deep into the caching system, but how do you want to do it?

Reply:
https://github.com/hurshi/dio-http-cache

Reply:
Well if we only use it for section and course searching then we probably don't need it. If we end up using it for the feed though, we could invalidate the cache when a user subscribes to a section. That said, we'd probably need a more elaborate invalidation plan.

For now, I'd just change that value to like an hour.